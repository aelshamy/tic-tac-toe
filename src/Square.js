import PropTypes from "prop-types";
import React from "react";

const Square = props => (
  <button className="square" onClick={props.onClick}>
    {props.value}
  </button>
);

Square.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func
};

export default Square;
