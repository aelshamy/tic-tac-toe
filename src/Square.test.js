import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import { create } from "react-test-renderer";
import Square from "./Square";

describe("<Sauare/>", () => {
  beforeEach(() => {});

  test("it will match the snapshot", () => {
    const c = create(<Square />);
    expect(toJson(c)).toMatchSnapshot();
  });

  describe("render()", () => {
    test("should render correctly", () => {
      const component = shallow(<Square />);
      expect(component.exists()).toBe(true);
      expect(component.find("button").exists()).toBe(true);
      expect(component.find("button").props().className).toEqual("square");
    });

    test("should rendered with the correct props", () => {
      const onClickFunction = () => {};
      const component = shallow(<Square onClick={onClickFunction} value="X" />);
      expect(component.find("button").props().children).toEqual("X");
      expect(component.find("button").props().onClick).toEqual(onClickFunction);
    });

    test("should use the click function provided", () => {
      let clicked = false;
      const onClickFunction = () => {
        clicked = true;
      };
      const component = shallow(
        <Square onClick={onClickFunction} value="AnyValue" />
      );
      expect(component.find("button").props().children).toEqual("AnyValue");
      expect(component.find("button").props().onClick).toEqual(onClickFunction);
      component.find("button").simulate("click");
      expect(clicked).toEqual(true);
    });
  });
});
