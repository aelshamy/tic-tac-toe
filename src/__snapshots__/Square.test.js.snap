// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sauare/> it will match the snapshot 1`] = `undefined`;
